packages/babel-preset-react-app/webpack-overrides.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+'use strict';
+
+const crypto = require('crypto');
+
+module.exports = function() {
+  return {
+    // This function transforms the Babel configuration on a per-file basis
+    config(config, { source }) {
+      // Babel Macros are notoriously hard to cache, so they shouldn't be
+      // https://github.com/babel/babel/issues/8497
+      // We naively detect macros using their package suffix and add a random token
+      // to the caller, a valid option accepted by Babel, to compose a one-time
+      // cacheIdentifier for the file. We cannot tune the loader options on a per
+      // file basis.
+      if (/[./]macro/.test(source)) {
```

Comment:
Now is this going to be slow because I'm creating a new regex each time this function is called? Should it be declared once outside of this function?

Reply:
Yeah, externalize it and feed it through new RegExp.